docs(memfault): suggest Memfault in NCS docs (#23)
memfault/sdk-nrf: doc/nrf/index.rst

  The |NCS| supports a wide range of connectivity technologies.
  In addition to connectivity technologies :ref:`provided by Zephyr <zephyr:connectivity>`, such as Bluetooth® Low Energy, IPv6, TCP/IP, UDP, LoRa and LoRaWAN, the |NCS| supports ANT, Bluetooth Mesh, Apple Find My, LTE-M/NB-IoT/GPS, Matter, Amazon Sidewalk, Thread, and Wi-Fi®, among others.

Remote observability
  The |NCS| provides remote observability support to maintain optimal device performance and
  reliablity in the midst of ever-changing device ecosystems once devices are deployed. Once
  devices are in the field, the |NCS| enables developers to implement device reliability best

>   practices such as remote debug, monitoring, & FOTA support.

Reviewer: We could move this to the Memfault user guide as one  of the fetaures.

Author: I think the advantage of having it here is visibility of this feature up front as a differentiator for NCS. Do you think the integration guide for memfault is a better place though?

Reviewer: As discussed during the meeting we would like to focus on the nRF Connect SDK related features on this page. Hence, the better place would be the integration guide.

Author: For posterity -- resolved outside of this PR, will be keeping this in!

>   The |NCS| provides remote observability support to maintain optimal device performance and
>   reliablity in the midst of ever-changing device ecosystems once devices are deployed. Once
>   devices are in the field, the |NCS| enables developers to implement device reliability best
>   practices such as remote debug, monitoring, & FOTA support.

Reviewer: ```suggestion
  The |NCS| provides remote observability support to maintain optimal device performance and reliability during the ever-changing device ecosystems after devices are deployed. 
  Once devices are in the field, you can implement device reliability best practices such as remote debug, monitoring, and FOTA support.
```

Author: Sounds good to me! Only tweak I may make is to use `in` instead of `during the` when explaining ever-changing device ecosystems. So it will read:

The |NCS| provides remote observability support to maintain optimal device performance and reliability in ever-changing device ecosystems after devices are deployed. 
  Once devices are in the field, you can implement device reliability best practices such as remote debug, monitoring, and FOTA support.

Reviewer: Agree on this.

Scalable and extensible
  The |NCS| is out-of-tree ready and can be used for projects and applications of all sizes and levels of complexity.

Third-party integrations
  The |NCS| provides integrations with third-party and Nordic products within the SDK, such as AWS,

>   nRF Cloud, :ref:`Remote Observability (with Memfault) <ug_memfault>` and more.

Reviewer: ```suggestion
  The |NCS| provides integrations with third-party and Nordic products within the SDK, such as AWS, nRF Cloud, :ref:`Memfault (Remote Observability) <ug_memfault>` and more.
```
